src/chapters/setup-intro-data-types/alternate-code-practice.rst

```diff
@@ -0,0 +1,406 @@
+
+            input = new Scanner(System.in);
+            System.out.println("Please enter your first name: ");
+            String name = input.next(); //Declares the 'name' variable and initializes it with text from the command line.
+            input.close();
+
+            System.out.println("Hello, " + name + "!");
+
+            // System.out.println("This line will NOT print!");
+         }
+      }
+
+Collect Input with the ``Scanner`` Class
+-----------------------------------------
+
+In Java, ``Scanner`` objects make getting input from the user, a file, or even
+over the network relatively easy. For our temperature conversion program, we
+declared the variable ``input`` to be of type ``Scanner``.
+
+.. sourcecode:: java
+   :lineno-start: 7
+
+   Scanner input;
+
+We want our program to prompt the user to enter in a number in the command
+line. We accomplish this by creating a ``Scanner`` instance using the word
+``new`` and then calling the constructor and passing it the ``System.in``
+object:
+
+.. sourcecode:: java
+   :lineno-start: 9
+
+   input = new Scanner(System.in);
+
+Notice that this ``Scanner`` object is assigned to the name ``input``, which we
+declared to be a ``Scanner`` object earlier. ``System.in`` is similar to
+``System.out`` except, as the name implies, it is used for input.
+
+.. admonition:: Note
+
+   If you are wondering why we must create a ``Scanner`` object to read data from
+   ``System.in`` when we can write data directly to ``System.out`` using
+   ``println``, you are not alone. We will talk about the reasons why this is so
+   when we dive into Java streams.
+
+Next, line 10 asks the user to enter a number, and in line 11 we use ``input``
+to read the value from the command line:
+
+.. sourcecode:: java
+   :lineno-start: 10
+
+   System.out.println("Enter the temperature in Fahrenheit: ");
+   fahrenheit = input.nextDouble();
+
+Here again we see the implications of Java being a strongly typed language.
+Notice that we must call the method ``nextDouble``, because the variable
+``fahrenheit`` was declared as a ``double``.
+
+Because Java is a statically typed language, we must call the appropriate
+method on the ``Scanner`` object to ensure the entered data is of the correct
+type. In this case, the compiler compares the types for ``fahrenheit`` and
+``input.nextDouble()`` and throws an error if the two do not match.
+
+The table below shows some commonly used methods of the ``Scanner`` class.
+There are many others supported by this class, and we will talk about how to
+find them in the next chapter.
+
+   TODO: check these descriptions for primitive vs object type usage??
+
+.. list-table:: ``Scanner`` methods
+   :header-rows: 1
+
+   * - Method Name
+     - Return Type
+     - Description
+   * - ``hasNext()``
+     - ``boolean``
+     - Returns ``true`` if more data is present.
+   * - ``hasNextInt()``
+     - ``boolean``
+     - Returns ``true`` if the next item to read is an integer.
```

Comment:
``integer``?

Reply:
Fixed to "Returns ``true`` if the next item to read is an ``int`` data type."